packages/xo-server/src/api/xostor.mjs

```diff
@@ -131,83 +145,91 @@ export const create = defer(async function (
   $defer,
   { description, disksByHost, force, ignoreFileSystems, name, provisioning, replication }
 ) {
-  const hostIds = Object.keys(disksByHost)
+  const task = await this.tasks.create({ name: `creation of XOSTOR: ${name}`, type: 'xo:xostor:create' })
+  return task.run(async () => {
+    const hostIds = Object.keys(disksByHost)
 
-  const tmpBoundObjectId = `tmp_${hostIds.join(',')}_${Math.random().toString(32).slice(2)}`
+    const tmpBoundObjectId = `tmp_${hostIds.join(',')}_${Math.random().toString(32).slice(2)}`
 
-  const xostorLicenses = await this.getLicenses({ productType: 'xostor' })
+    const license = await Task.run({ properties: { name: 'license check' } }, async () => {
+      const xostorLicenses = await this.getLicenses({ productType: 'xostor' })
 
-  const now = Date.now()
-  const availableLicenses = xostorLicenses.filter(
-    ({ boundObjectId, expires }) => boundObjectId === undefined && (expires === undefined || expires > now)
-  )
+      const now = Date.now()
+      const availableLicenses = xostorLicenses.filter(
+        ({ boundObjectId, expires }) => boundObjectId === undefined && (expires === undefined || expires > now)
+      )
 
-  let license = availableLicenses.find(license => license.productId === 'xostor')
+      let _license = availableLicenses.find(({ productId }) => productId === 'xostor')
```

Comment:
Why not keep `license`?

Both variables exist in different scopes and represent both the same value.

Reply:
This was just to avoid any confusion.

Reply:
IMHO it makes the code more confusing than less, but it's my personal opinion, do as you like.